src/settings-tab.ts

```diff
@@ -112,22 +112,36 @@ export class MindMapSettingsTab extends PluginSettingTab {
           })
       );
 
+    function decide_display_colors(approach: MindMapSettings['coloring']) {
+      const all = [color_1, color_2, color_3, color_default]
+      const settings = {
+        branch: [],
+        depth: all,
+        single: [color_default]
+      }
+      all.forEach(setting => setting.settingEl.hidden = true)
+      settings[approach].forEach(setting => setting.settingEl.hidden = false)
+    }
+
     new Setting(containerEl)
-      .setName("Coloring approach")
-      .setDesc(
-        "The 'depth' changes the color on each level, 'branch' changes the color on each new branch"
-      )
-      .addDropdown((dropDown) =>
-        dropDown
-          .addOption("depth", "Depth based coloring")
-          .addOption("branch", "Branch based coloring")
-          .setValue(this.plugin.settings.coloring || "depth")
-          .onChange((value: "branch" | "depth") => {
-            this.plugin.settings.coloring = value;
-            save();
-          })
-      );
+    .setName("Coloring approach")
+    .setDesc(
+      "The 'depth' changes the color on each level, 'branch' changes the color on each new branch"
+    )
+    .addDropdown((dropDown) =>
+      dropDown
+        .addOption("depth", "Depth based coloring")
+        .addOption("branch", "Branch based coloring")
+        .addOption("single", "Single color")
+        .setValue(this.plugin.settings.coloring || "depth")
+        .onChange((value: "branch" | "depth" | "single") => {
+          this.plugin.settings.coloring = value;
+          save();
+          decide_display_colors(value);
+        })
+    );
 
+    const color_1 =
     new Setting(containerEl)
```

Comment:
- move this new Setting to the previous line, right after the `=`, followed by a space char.